src/Core/Parsers/ODataASTCosmosVisitor.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.OData.UriParser;
+
+/// <summary>
+/// This class is a visitor for an AST generated when parsing a $filter query string
+/// with the OData Uri Parser for Cosmos DB Db Policies
+/// </summary>
```

Comment:
Right now, only difference is, Cosmos Visitor is simple, just need "prefix" information to get appended to column,  but SQL visitor needs more information.
Second thing, tomorrow, you might want to handle different other kind of queries differently for cosmos and sql, this would give you this flexibility.